sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_patch.py

```diff
@@ -0,0 +1,724 @@
+        :return: The SchemaProperties associated with the provided schema metadata.
+        :rtype: ~azure.schemaregistry.SchemaProperties
+        :raises: :class:`~azure.core.exceptions.HttpResponseError`
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/sync_samples/sample_code_schemaregistry.py
+                :start-after: [START get_schema_id_sync]
+                :end-before: [END get_schema_id_sync]
+                :language: python
+                :dedent: 4
+                :caption: Get schema id.
+
+        """
+        format = get_case_insensitive_format(format)
+        http_request_kwargs = get_http_request_kwargs(kwargs)
+        # ignoring return type because the generated client operations are not annotated w/ cls return type
+        schema_properties: Dict[
+            str, Union[int, str]
+        ] = self._generated_client.get_schema_id_by_content(  # type: ignore
+            group_name=group_name,
+            name=name,
+            schema_content=cast(IO[Any], definition),
+            content_type=kwargs.pop("content_type", get_content_type(format)),
+            cls=partial(prepare_schema_properties_result, format),
+            **http_request_kwargs,
+        )
+        return SchemaProperties(**schema_properties)
+
+
+class SchemaProperties(object):
+    """
+    Meta properties of a schema.
+
+    :ivar id: References specific schema in registry namespace.
+    :vartype id: str
+    :ivar format: Format for the schema being stored.
+    :vartype format: ~azure.schemaregistry.SchemaFormat
+    :ivar group_name: Schema group under which schema is stored.
+    :vartype group_name: str
+    :ivar name: Name of schema.
+    :vartype name: str
+    :ivar version: Version of schema.
+    :vartype version: int
+    """
+
+    def __init__(self, **kwargs: Any) -> None:
+        self.id = kwargs.pop("id")
+        self.format = kwargs.pop("format")
+        self.group_name = kwargs.pop("group_name")
+        self.name = kwargs.pop("name")
+        self.version = kwargs.pop("version")
+
+    def __repr__(self):
+        return (
+            f"SchemaProperties(id={self.id}, format={self.format}, "
+            f"group_name={self.group_name}, name={self.name}, version={self.version})"[
+                :1024
+            ]
+        )
+
+
+class Schema(object):
+    """
+    The schema content of a schema, along with id and meta properties.
+
+    :ivar definition: The content of the schema.
+    :vartype definition: str
+    :ivar properties: The properties of the schema.
+    :vartype properties: SchemaProperties
+    """
+
+    def __init__(self, **kwargs: Any) -> None:
+        self.definition = kwargs.pop("definition")
+        self.properties = kwargs.pop("properties")
+
+    def __repr__(self):
+        return f"Schema(definition={self.definition}, properties={self.properties})"[
+            :1024
+        ]
+
+
+class SchemaFormat(str, Enum, metaclass=CaseInsensitiveEnumMeta):
```

Comment:
Curious why these models arent part of the generated code?

Reply:
hmm, no reason other than I didn't know we could/wanted to do that, since we were previously adding these models manually. But if we could generate them, that would be great!

Reply:
actually, one reason to not generate this SchemaFormat model specifically is b/c different languages use different casing for the enum values here. "Json" vs. "JSON". I think technically, all the clients should be doing case insensitive checks here anyway, so maybe we still could generate it? 🤔

Reply:
@swathipil and I synced on this offline. It's better to add the models into the tsp and do this migration once. If some of the SDKs made larger changes to the service model schemas then they can use the patch mechanisms to get  to where they were before. For models and operations where there were simple renames, `@projectedName` can be used to help with SDK generation. These customizations should be added in client.tsp.